crates/sui-core/src/transaction_orchestrator.rs

```diff
@@ -214,4 +228,56 @@ where
     ) -> tokio::sync::broadcast::Receiver<(CertifiedTransaction, CertifiedTransactionEffects)> {
         self.quorum_driver_handler.subscribe()
     }
+
+    /// Execute a finalized transaction locally.
+    /// Firstly it tries to execute it optimistically. If there are missing
+    /// dependencies, it then leverages Node Sync to process the parents.
+    async fn execute_impl(
+        state: &Arc<AuthorityState>,
+        node_sync_handle: &NodeSyncHandle,
+        tx_cert: &CertifiedTransaction,
+        effects_cert: &CertifiedTransactionEffects,
+    ) -> SuiResult {
+        let tx_digest = tx_cert.digest();
+        let res = state
+            .handle_certificate_with_effects(tx_cert, effects_cert)
+            .await;
+        match res {
+            Ok(_) => {
+                debug!(
+                    ?tx_digest,
+                    "Orchestrator optimistically executed transaction successfully."
+                );
+                Ok(())
+            }
+            Err(SuiError::ObjectNotFound { .. }) | Err(SuiError::ObjectErrors { .. }) => {
```

Comment:
@mystenmark  I think `SharedObjectLockNotSet` is not relevant here because `handle_certificate_with_effects` sets the locks first before checking them.